nilearn/plotting/displays.py

```diff
@@ -884,15 +933,28 @@ def annotate(self, left_right=True, positions=True, size=12, **kwargs):
                 kwargs['color'] = 'k'
 
         bg_color = ('k' if self._black_bg else 'w')
+
+        # Cleanup scale bar's kwargs
+        scale_bar_kwargs = {
+            key: kwargs.pop(key) for key in ['scale_width', 'units']
+            if key in kwargs}
+
         if left_right:
-            for display_ax in self.axes.values():
-                display_ax.draw_left_right(size=size, bg_color=bg_color,
-                                           **kwargs)
+            for display_axis in self.axes.values():
+                display_axis.draw_left_right(size=size, bg_color=bg_color,
+                                             **kwargs)
 
         if positions:
-            for display_ax in self.axes.values():
-                display_ax.draw_position(size=size, bg_color=bg_color,
-                                         **kwargs)
+            for display_axis in self.axes.values():
+                display_axis.draw_position(size=size, bg_color=bg_color,
+                                           **kwargs)
+
+        if scalebar:
+            kwargs.update(scale_bar_kwargs)  # Insert kwargs back
```

Comment:
kwargs should not be passed to draw_scale_bar. kwargs should be passed to only one type of function. If it's really necessary to route arguments down to this function, they should be added explicitly to the list of arguments of the calling function.

I am not convinced that extra arguments for the scalebar need to be passed around. Users can always use the specific function.
I am not convinced that it is